### DISCUSSION TITLE
PixelCNN for PixelDefend

### DISCUSSION THREAD

**Discussion by MiguelCosta94** [Trusted-AI/adversarial-robustness-toolbox#1921] [Q&A]
Hi,
While experimenting with PixelDefend defense I spotted that PixelDefend expects a PixelCNN as an argument. The original implementation ( https://arxiv.org/abs/1710.10766 ) of PixelDefend works with PixelCNN++. Is your implementation expecting a PixelCNN++ or the normal PixelCNN?

**Comment by beat-buesser** [comment]
Hi @MiguelCosta94 I think `PixelCNN` is a modification of `PixelCNN++` and it looks like they are required to reproduce the paper.
